src/state-mutation.js

```diff
@@ -121,12 +121,16 @@ export const updateOrInsertResource = (state, resource) => {
     const idx = resources.findIndex(item => item.id === resource.id);
 
     const relationships = {};
-    for (const relationship in resource.relationships) {
-      if (!resource.relationships[relationship].data) {
+    for (const relationship in resources[idx].relationships) {
+      if (!resource.relationships || !resource.relationships[relationship] || !resource.relationships[relationship].data) {
```

Comment:
We have a small utility method [`hasOwnProperties`](https://github.com/stonecircle/redux-json-api/blob/master/src/utils.js#L25) which can make this a bit more readable 🙂